model/task/db.go

```diff
@@ -2135,6 +2135,60 @@ func GetTasksByVersion(ctx context.Context, versionID string, opts GetTasksByVer
 	return results, count, nil
 }
 
+// GetTaskStatusesByVersion gets all unique task display statuses for a specific version
+func GetTaskStatusesByVersion(ctx context.Context, versionID string) ([]string, error) {
+
+	opts := GetTasksByVersionOptions{
+		IncludeBaseTasks:               false,
+		FieldsToProject:                []string{DisplayStatusKey},
+		IncludeBuildVariantDisplayName: false,
+		IncludeNeverActivatedTasks:     true,
+		IncludeExecutionTasks:          false,
+	}
+	pipeline, err := getTasksByVersionPipeline(versionID, opts)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "getting tasks by version pipeline")
+	}
+
+	pipeline = append(pipeline, bson.M{
+		"$group": bson.M{
+			"_id": nil,
+			"statuses": bson.M{
+				"$addToSet": "$" + DisplayStatusKey,
+			},
+		},
+	})
+	pipeline = append(pipeline, bson.M{
```

Comment:
What's the goal behind having these sorted?

Reply:
It ensures they are returned in a consistent order otherwise the order changes every iteration which makes testing this impossible.